### PR TITLE
Add new repositories to a highlighted user defined list

### PR DESCRIPTION
**Pull request by norascheuch**
When a remote user defined list is highlighted when pressing the 'Add new Database'-Button we only want to show the option to add a new repository (instead of also adding a new owner) and add the new database to the highlighted list.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/vscode-tests/minimal-workspace/databases/db-panel.test.ts
Comment (norascheuch) [inline extensions/ql-vscode/src/vscode-tests/minimal-workspace/databases/db-panel.test.ts:472]: These tests were copied from `describe("addNewList",`. I'm concerned we're calling the `dbManager` directly instead of using the `db-panel` (as it's a db-panel test).

Is it possible that we should move these tests into the dbManager-test-file?..

File: extensions/ql-vscode/src/vscode-tests/minimal-workspace/databases/db-panel.test.ts
Comment (norascheuch) [inline extensions/ql-vscode/src/vscode-tests/minimal-workspace/databases/db-panel.test.ts:514]: See my comment above. Also: to call the `db-panel` directly we need to stub the `private async getHighlightedDbItem()`-method on the panel. Any ideas how we can achieve that?

File: extensions/ql-vscode/src/databases/ui/db-panel.ts
Comment (shati-patel) [inline extensions/ql-vscode/src/databases/ui/db-panel.ts:89]: From the (internal) issue: 
> If a list or item inside a list is highlighted, add it to that list

So I think that means something like this? (i.e. also checking if the highlighted item is inside a list)
```suggestion
    if (highlightedItem?.kind === DbItemKind.RemoteUserDefinedList) {
      await this.addNewRemoteRepo(highlightedItem.listName);
    } else if (
      highlightedItem?.kind === DbItemKind.RemoteRepo &&
      highlightedItem.parentListName
    ) {
      await this.addNewRemoteRepo(highlightedItem.parentListName);
    } else {
```

File: extensions/ql-vscode/src/vscode-tests/minimal-workspace/databases/db-panel.test.ts
Comment (shati-patel) [reply extensions/ql-vscode/src/vscode-tests/minimal-workspace/databases/db-panel.test.ts:518]: I think we weren't able to test/stub the `db-panel` methods directly, so that's why we're testing the `db-manager` methods instead 😬

File: extensions/ql-vscode/src/vscode-tests/minimal-workspace/databases/db-panel.test.ts
Comment (shati-patel) [reply extensions/ql-vscode/src/vscode-tests/minimal-workspace/databases/db-panel.test.ts:476]: You're right: this test file doesn't actually test the panel, rather the manager, config store, tree data provider, and the interactions between them 🤔 

It probably makes sense to restructure/rename this test file, but that might be a question for a future PR 😅 Happy to keep this as-is for now.

**Review by shati-patel — COMMENTED** [review]
Great! Thanks for adding this ⚡  One comment, and I tried answering your questions (though you might want a more informed opinion 😅)

File: extensions/ql-vscode/src/databases/ui/db-panel.ts
Comment (norascheuch) [reply extensions/ql-vscode/src/databases/ui/db-panel.ts:89]: Oh yes, didn't think about that! Will add the case!

File: extensions/ql-vscode/src/databases/db-manager.ts
Comment (charisk) [inline extensions/ql-vscode/src/databases/db-manager.ts:82]: I think these checks should happen in the `DbConfigStore` if they aren't already, and the `DbManager` should just pass through to that. That's what we did elsewhere.

File: extensions/ql-vscode/src/databases/config/db-config-store.ts
Comment (charisk) [inline extensions/ql-vscode/src/databases/config/db-config-store.ts:117]: This is fine, but it will keep looping even after it found the list, which is a little strange. I don't expect it to be a problem because we don't anticipate large numbers of lists.

We could instead find the list with the given name, and update its `repositories`.

Comment (charisk) [inline extensions/ql-vscode/src/databases/config/db-config-store.ts:117]: Shall we also check whether the parent list exists at all and throw an error if it doesn't?

**Review by charisk — APPROVED** [review]
Good stuff, just some suggestions!

File: extensions/ql-vscode/src/databases/db-manager.ts
Comment (norascheuch) [reply extensions/ql-vscode/src/databases/db-manager.ts:82]: Hm, this must've sneaked in during a rebase. Great..

File: extensions/ql-vscode/src/databases/config/db-config-store.ts
Comment (norascheuch) [reply extensions/ql-vscode/src/databases/config/db-config-store.ts:117]: Hm, in which case could it not exist?.. We can check for it though.

Comment (norascheuch) [reply extensions/ql-vscode/src/databases/config/db-config-store.ts:117]: 👍

Comment (charisk) [reply extensions/ql-vscode/src/databases/config/db-config-store.ts]: It's one of those things that _should never_ happen.

**Review by charisk — APPROVED** [review]
LGTM! 🎉

**Review by shati-patel — APPROVED** [review]
Not sure if you were waiting for another review from me - all looks good :)

**Pull request merged**
